cajita/src/Cajita_DynamicPartitioner.hpp

```diff
@@ -27,16 +27,31 @@
 
 namespace Cajita
 {
+
+template <typename Device>
+class DynamicPartitionerWorkloadMeasurer
+{
+    using memory_space = typename Device::memory_space;
+
+  public:
+    virtual void compute( Kokkos::View<int***, memory_space>& ) = 0;
+};
+
 //---------------------------------------------------------------------------//
 /*!
-  Sparse mesh block partitioner. (Current Version: Support 3D only)
+  Dynamic mesh block partitioner. (Current Version: Support 3D only) There
+  should be no instantiation for this class without implementing any workload
+  computation.
+
   \tparam Device Kokkos device type.
-  \tparam CellPerTileDim Cells per tile per dimension.
-  \tparam NumSpaceDim Dimemsion (The current version support 3D only)
+  \tparam CellPerTileDim Cells
+  per tile per dimension.
+  \tparam NumSpaceDim Dimemsion (The current version
+  support 3D only)
 */
 template <typename Device, unsigned long long CellPerTileDim = 4,
```

Comment:
This really should be templated on the memory space instead of device type, but I'm hesitant to add another change in this PR